core/domain/topic_domain.py

```diff
@@ -347,22 +500,158 @@ def update_additional_story_ids(self, new_additional_story_ids):
         """
         self.additional_story_ids = new_additional_story_ids
 
-    def update_skill_ids(self, new_skill_ids):
+    def add_uncategorized_skill_id(self, new_uncategorized_skill_id):
         """Updates the skill id list of a topic object.
 
         Args:
-            new_skill_ids: list(str). The updated list of skill ids.
+            new_uncategorized_skill_id: str. The new skill id to add to
+                uncategorized_skill_ids list.
+
+        Raises:
+            Exception. The given skill id is already present in a subtopic.
+        """
+        for subtopic in self.subtopics:
+            if new_uncategorized_skill_id in subtopic.skill_ids:
+                raise Exception(
+                    'The skill id %s already exists in subtopic with id %s.'
+                    % (new_uncategorized_skill_id, subtopic.id))
+        self.uncategorized_skill_ids.append(new_uncategorized_skill_id)
```

Comment:
Should also check that it's not in the uncategorized_skill_ids list, right?

Reply:
Done.